maven-extension/src/main/java/io/opentelemetry/maven/ChainedExecutionListener.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.maven;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.maven.execution.ExecutionEvent;
+import org.apache.maven.execution.ExecutionListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util class to chain multiple {@link ExecutionListener} as Maven APIs don't offer this capability.
+ */
+final class ChainedExecutionListener implements ExecutionListener {
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  private final List<ExecutionListener> listeners;
+
+  /** @param listeners {@code null} values are filtered */
+  ChainedExecutionListener(ExecutionListener... listeners) {
+    this.listeners = Arrays.stream(listeners).filter(e -> e != null).collect(Collectors.toList());
+  }
+
+  @Override
+  public void projectDiscoveryStarted(ExecutionEvent event) {
+    for (ExecutionListener listener : this.listeners) {
+      listener.projectDiscoveryStarted(event);
+    }
+  }
+
+  @Override
+  public void sessionStarted(ExecutionEvent event) {
+    for (ExecutionListener listener : this.listeners) {
+      try {
+        listener.sessionStarted(event);
+      } catch (RuntimeException e) {
+        logger.error("Silently skip exception", e);
```

Comment:
```suggestion
        logger.error("Error starting session", e);
```

It's clearly not silent if we're logging :P